verifiers/internal/gha/verifier.go

```diff
@@ -43,68 +44,79 @@ func verifyEnvAndCert(env *dsse.Envelope,
 	provenanceOpts *options.ProvenanceOpts,
 	builderOpts *options.BuilderOpts,
 	defaultBuilders map[string]bool,
-) ([]byte, string, error) {
+) ([]byte, *utils.BuilderID, error) {
 	/* Verify properties of the signing identity. */
 	// Get the workflow info given the certificate information.
 	workflowInfo, err := GetWorkflowInfoFromCertificate(cert)
 	if err != nil {
-		return nil, "", err
+		return nil, nil, err
 	}
 
 	// Verify the workflow identity.
 	builderID, err := VerifyWorkflowIdentity(workflowInfo, builderOpts,
 		provenanceOpts.ExpectedSourceURI, defaultBuilders)
 	if err != nil {
-		return nil, "", err
+		return nil, nil, err
 	}
 
 	// Verify properties of the SLSA provenance.
 	// Unpack and verify info in the provenance, including the Subject Digest.
 	provenanceOpts.ExpectedBuilderID = builderID
 	if err := VerifyProvenance(env, provenanceOpts); err != nil {
-		return nil, "", err
+		return nil, nil, err
 	}
 
 	fmt.Fprintf(os.Stderr, "Verified build using builder https://github.com%s at commit %s\n",
 		workflowInfo.JobWobWorkflowRef,
 		workflowInfo.CallerHash)
 	// Return verified provenance.
 	r, err := base64.StdEncoding.DecodeString(env.Payload)
-	return r, builderID, err
+	if err != nil {
+		return nil, nil, err
+	}
+
+	// Temporary code.
+	bid, _ := utils.BuilderIDNew(builderID)
```

Comment:
if someone supplies their own malformed builder ID in here, this could fail right?

Reply:
I dont think so, because GHA only matches on the builder name, not the version. This is a temporary hack: in next PR I will add support for GHA

Reply:
actually I removed the error checking, otherwise it always fails since the `builderID` never contains the `@v1.2.3` on GHA. Adding support in the next PR.